rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2194,6 +2195,218 @@ void save_callsStorageProviderSave() {
         verify(storageProvider).save(federationMainnetConstants.getBtcParams(), activations);
     }
 
+    @Nested
+    @Tag("commit federation")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class CommitFederationTests {
+        private ActivationConfig.ForBlock activations;
+
+        private FederationStorageProvider federationStorageProvider;
+        private FederationSupportImpl federationSupport;
+        private PendingFederation pendingFederation;
+        private BridgeEventLogger bridgeEventLogger;
+        
+        private Block rskExecutionBlock;
+        long rskExecutionBlockNumber = 1000L;
+        long rskExecutionBlockTimestamp = 10L;
+
+        @BeforeEach
+        void setUp() {
+            StorageAccessor bridgeStorageAccessor = new InMemoryStorage();
+            federationStorageProvider = new FederationStorageProviderImpl(bridgeStorageAccessor);
+            
+            pendingFederation = new PendingFederation(FederationTestUtils.getFederationMembers(5));
+            federationStorageProvider.setPendingFederation(pendingFederation);
+
+            bridgeEventLogger = mock(BridgeEventLogger.class);
+
+            rskExecutionBlock = mock(Block.class);
+            when(rskExecutionBlock.getNumber()).thenReturn(rskExecutionBlockNumber);
+            when(rskExecutionBlock.getTimestamp()).thenReturn(rskExecutionBlockTimestamp);
+        }
+
+        @Test
+        void commitPendingFederationAccordingToActivations_preRSKIP419_shouldCallPreRSKIP419Method() {
```

Comment:
Maybe call it preLovell instead of preRskipX?

We should standardize that

Reply:
Also, not sure this test is a good idea.

We should be testing behaviour, not the internals of a class. Instead of spying federationSupport object I think we should simply call commitFederation pre RSKIP and assert whatever is different from when it's called post RSKIP

Reply:
Similar situation with the test names. Rather than `shouldCallXMethod` I think it should be `shouldDoX`

Reply:
i removed those two tests.

Regarding the test name, preLovell or preRskip, I found putting the rskip better. I think is more accurate, since he behavior changes because of the rskip activation